plugins/ssrmatching/ssrmatching.ml

```diff
@@ -354,28 +354,30 @@ let nf_open_term sigma0 ise c =
   let c' = nf c in let _ = Evd.fold copy_def sigma0 () in
   !s', Evd.evar_universe_context s, EConstr.of_constr c'
 
-let unif_end env sigma0 ise0 pt ok =
+let unif_end ?(solve_TC=true) env sigma0 ise0 pt ok =
```

Comment:
Are you sure you want the flag to be optional? Shouldn't we look at each call and decide whether it makes sense to resolve TCs or not?

Reply:
If you want to have all ssr code be double checked for compatibility with TC, then yes, but it quite more ambitious.

I think that the current code is a bit messy, in particular `_end` means very little.
Historically `unif_end` was called at the end of a matching operation that also corresponded to the end of a tactic, and there were no TC. Then I started to use it as part as a higher level API unify_HO (since it ensure all remaining unif problems are solved, eg that unification did work for real). 
Then I think TC were added and it still makes sense to resolve TC at the end of a tactic (since the lemma you use may need these classes). But I don't think solving TC makes sense after a unification step.

After CI finishes (the failure seems unrelated, and works here), I'm pushing a commit with two different defaults:
- on for unif_end
- off for unify_HO
and let's see how it goes.